android/src/main/java/com/tailscale/ipn/IPNReceiver.java

```diff
@@ -0,0 +1,20 @@
+package com.tailscale.ipn;
+
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+
+public class IPNReceiver extends BroadcastReceiver {
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        if (intent.getAction() == "com.tailscale.ipn.CONNECT_VPN"){ //  android.intent.action.CONNECT_VPN
```

Comment:
formatting: space before final { (and on line 13, which needs a spaces around `else if` too)

Reply:
No worries, made those changes